contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -8707,6 +8879,23 @@ handleClrUdtFuncCall(TSqlParser::Clr_udt_func_callContext *ctx)
 				else ind = method_calls[i-1]->stop->getStopIndex();
 				rewrite_geospatial_query_helper(ctx, method, ind);
 			}
+
+			/* rewriting the query in case of xml function calls */
+			if(method->xml_method_call())
+			{
+				size_t ind = -1;
+				if (i == 0)
+				{
+					if(ctx->local_id()) ind = ctx->local_id()->stop->getStopIndex();
+					else if(ctx->subquery()) ind = ctx->subquery()->stop->getStopIndex();
+					else if(ctx->function_call()) ind = ctx->function_call()->stop->getStopIndex();
+					else if(ctx->RR_BRACKET()) ind = ctx->RR_BRACKET()->getSymbol()->getStopIndex();
+				}
+				else ind = method_calls[i-1]->stop->getStopIndex();
```

Comment:
Code duplication with clr_udt handling, please have a common code block for these

Reply:
refactored code to have a common code block.